pennylane/transforms/split_qscript.py

```diff
@@ -338,28 +229,33 @@ def sum_expand(qscript: QuantumScript, group=True):
                 tmp_idxs.append([idxs_coeffs[measurements.index(m)] for m in m_group])
         idxs_coeffs = tmp_idxs
         qscripts = [
-            QuantumScript(ops=qscript._ops, measurements=m_group, prep=qscript._prep)
+            qscript.__class__(ops=qscript._ops, measurements=m_group, prep=qscript._prep)
             for m_group in m_groups
         ]
     else:
         qscripts = [
-            QuantumScript(ops=qscript._ops, measurements=[m], prep=qscript._prep)
+            qscript.__class__(ops=qscript._ops, measurements=[m], prep=qscript._prep)
             for m in measurements
         ]
 
     def processing_fn(expanded_results):
         results = []  # [(m_idx, result)]
         for qscript_res, qscript_idxs in zip(expanded_results, idxs_coeffs):
             if isinstance(qscript_idxs[0], tuple):  # qscript_res contains only one result
-                if not qml.active_return():  # old return types
+                if not qml.active_return() and len(qscript_res) == 1:  # old return types
                     qscript_res = qscript_res[0]
                 for idx, coeff in qscript_idxs:
-                    results.append((idx, qscript_res if coeff is None else coeff * qscript_res))
+                    res = qml.math.convert_like(qml.math.dot(coeff, qscript_res), qscript_res)
+                    results.append((idx, res))
                 continue
             # qscript_res contains multiple results
-            for res, idxs in zip(qscript_res, qscript_idxs):
+            if qml.active_return():
+                qscript_res = qml.math.stack(qscript_res)
```

Comment:
Just to double because I think it is right: qscript_res only contains expectation values, and therefore stacking here is not an issue?

Reply:
Nope, we can have multiple values here...

Thus it might happen that we stack dictionaries (`qml.counts`) with lists (`qml.probs`) and scalars (`qml.expval`)...

I refactored this to avoid using stack. Let me know!!